# Profitable App Profiles for the App Store and Google Play Markets

We only build apps that are free to download and install, and our main source of revenue consists of in-app ads. This means that the number of users of our apps determines our revenue for any given app — the more users who see and engage with the ads, the better. Our goal for this project is to analyze data to help our developers understand what type of apps are likely to attract more users.

## Opening and Exploring datasets

- [A dataset](https://www.kaggle.com/lava18/google-play-store-apps) containing data about approximately 10,000 Android apps from Google Play; the data was collected in August 2018. You can download the dataset directly from this link.
- [A dataset](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps) containing data about approximately 7,000 iOS apps from the App Store; the data was collected in July 2017. You can download the dataset directly from this link.

First of all, let's open files 'AppleStore.csv' and 'googleplaystore.csv'.

In [1]:
from csv import reader 

def open_dataset(file_name, header = True) :
    '''
    This function returns dataset and header of file.
    We can divide header of file by using header = True. 
    '''
    opened_file = open(file_name)
    read_file = reader(opened_file)
    data = list(read_file)
    
    if header :
        return data[1:], data[0]
    else : 
        return data

In [2]:
# Extract header and data from files 

apple_apps, apple_header = open_dataset('AppleStore.csv')
android_apps, android_header = open_dataset('googleplaystore.csv')

In [3]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

In [4]:
# Explore 'AppleStore.csv'

print(f'Header of AppleStore : {apple_header}\n')
print(explore_data(apple_apps, 0, 3, rows_and_columns = True))

Header of AppleStore : ['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows: 7197
Number of columns: 16
None


Apple store dataset have 7197 rows and 16 columns. We can use 'track_name', 'price', 'user_rating', 'prime_genre' for analysis. 

In [5]:
# Explore 'googleplaystore.csv'

print(f'Header of googleplay : {android_header}\n')
print(explore_data(android_apps, 0, 3, rows_and_columns = True))

Header of googleplay : ['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 10841
Number of columns: 13
None


In Googleplay store, dataset have 10841 rows and 13 columns. We can use 'App', 'Category', Rating', 'Price', 'Genres' for analysis.

## Detecting wrong data

We only build apps that are free to download and install, and we design them for an English-speaking audience.

There is one of the discussions describes an error for a certain row, 10472. It saids "Wrong Entry for 10472 row if Header is not included. Header: - ['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']
Row: - ['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']
Category is not present and hence column shift has happened in this row.

In [6]:
print(android_apps[10472]) 
del android_apps[10472]

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


In [7]:
print(android_apps[10472])

['osmino Wi-Fi: free WiFi', 'TOOLS', '4.2', '134203', '4.1M', '10,000,000+', 'Free', '0', 'Everyone', 'Tools', 'August 7, 2018', '6.06.14', '4.4 and up']


## Dropping duplicate data

If we explore the Google Play data set long enough or look at discussions section, we can notice some apps have duplicated entries. 

In [8]:
for app in android_apps : 
    name = app[0]
    if name == 'Instagram' : 
        print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


In [9]:
duplicate_apps = []
unique_apps = []

for app in android_apps : 
    name = app[0]
    
    if name in unique_apps :
        duplicate_apps.append(name)
    else : 
        unique_apps.append(name) 
        
print(f'Number of duplicate apps : {len(duplicate_apps)}')
print('\n')
print(f'Examples of duplicate apps : {duplicate_apps[:10]}')

Number of duplicate apps : 1181


Examples of duplicate apps : ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack']


In total, there are 1,181 cases when an app occurs more than one.

We don't want to count certain apps more than once, so we need to drop the duplicate data. In duplicate dataset apps' name is 'Instagram', there are only difference in reviews. By making certain criterion, we can drop duplicate value more easily. 

In [10]:
# Length of rows before remove duplicates

print(f'Total length : {len(android_apps)}')

# Length of rows after romove duplicates

print(f'Expected length : {len(android_apps) - len(duplicate_apps)}')

Total length : 10840
Expected length : 9659


To remove the duplicates, we will do the following : 

- Create a dictionary, where each dictionary key is a unique app name and the corresponding dictionary value is the highest number of reviews of that app
- Use the information stored in the dictionary and create a new dataset, which will have only one entry per app (and for each app, we'll only select the entry with the highest number of reviews) 

1. Create a dictionary where each key is a unique app name and the corresponding dictionary value is the highest number of reviews of that app.

- Start by creating an empty dictionary named reviews_max.
- Loop through the Google Play data set (don't include the header row). For each iteration, do the following:
    - Assign the app name to a variable named name.
    - Convert the number of reviews to float. Assign it to a variable named n_reviews.
    - If name already exists as a key in the reviews_max dictionary and reviews_max[name] < n_reviews, update the number of reviews for that entry in the reviews_max dictionary.
    - If name is not in the reviews_max dictionary as a key, create a new entry in the dictionary where the key is the app name, and the value is the number of reviews. Make sure you don't use an else clause here, or the number of reviews will be incorrectly updated whenever reviews_max[name] < n_reviews evaluates to False.
- Inspect the dictionary to make sure everything went as expected. Measure the length of the dictionary — remember that the expected length is 9,659 entries.



In [11]:
# Create a dictionary 

reviews_max = {}

for app in android_apps : 
    name = app[0]
    n_reviews = float(app[3])
    
    if (name in reviews_max) and (reviews_max[name] < n_reviews) :
        reviews_max[name] = n_reviews
    elif (name not in reviews_max) :
        reviews_max[name] = n_reviews 
        
print(reviews_max['Instagram'])
print('\n')
print(f'Length of reviews_max is {len(reviews_max)}.')

66577446.0


Length of reviews_max is 9659.


2. Use the dictionary you created above to remove the duplicate rows:

- Start by creating two empty lists: android_clean (which will store our new cleaned data set) and already_added (which will just store app names).
- Loop through the Google Play dataset (don't include the header row), and for each iteration, do the following:
    - Assign the app name to a variable named name.
    - Convert the number of reviews to float, and assign it to a variable named n_reviews.
- If n_reviews is the same as the number of maximum reviews of the app name (the number can be found in the reviews_max dictionary) and name is not already in the list already_added (read the solution notebook to find out why we need this supplementary condition):
    - Append the entire row to the android_clean list (which will eventually be a list of lists and store our cleaned dataset).
    - Append the name of the app name to the already_added list — this helps us to keep track of apps that we already added.



In [12]:
# Make new dataset removing duplicates 

android_clean = []    # store our new cleaned dataset
already_added = []    # store app names 

for app in android_apps : 
    name = app[0]
    n_reviews = float(app[3])
    
    if (n_reviews == reviews_max[name]) and (name not in already_added) : 
        android_clean.append(app)
        already_added.append(name)
    
print(f'Processed length : {len(android_clean)}')

Processed length : 9659


android_clena dataset have 9,659 rows same as expected rows. 

## Deleting Non English apps

If we explore the data long enough, we'll find theat both datasets have apps with names that suggest they are not designed for an English-speaking audience. 

We will use ASCII system to remove non-English apps using the number corresponding to the characters we commonly use in an English text are all in the range 0 to 127. Based on this number range, we can build a function that detects wheter a character belongs to the set of common English character or not.

1. Write a function that takes in a string and returns False if there's any character in the string that doesn't belong to the set of common English characters; otherwise, the function returns True.

- Inside the function, iterate over the input string. For each iteration check whether the number associated with the character is greater than 127. When a character is greater than 127, the function should immediately return False — the app name is probably non-English since it contains a character that doesn't belong to the set of common English characters.
- If the loop finishes running without the return statement executing, then it means no character had a corresponding number over 127 — the app name is probably English, so the functions should return True.



In [13]:
def english_detector(string) :
    
    for character in string : 
        if ord(character) > 127 : 
            return False
    return True
    
print(english_detector('Instagram'))
print(english_detector('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(english_detector('Docs To Go™ Free Office Suite'))
print(english_detector('Instachat 😜'))

True
False
False
False


But we saw that the function couldn't correctly identify certain English app names like 'Docs To Go™ Free Office Suite' and 'Instachat 😜'. 

To minimize the impact of data loss, we'll only remove an app if its name has more than three characters with corresponding numbers falling outside the ASCII range. 

In [14]:
def english_detector(string) :
    count_over = 0
    
    for character in string : 
        if ord(character) > 127 : 
            count_over += 1
            
    if count_over > 3 :
        return False 
    return True 

print(english_detector('Instagram'))
print(english_detector('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(english_detector('Docs To Go™ Free Office Suite'))
print(english_detector('Instachat 😜'))

True
False
True
True


We can see function works pretty well. 

In [15]:
print(f'Before length : {len(android_clean), len(apple_apps)}')

android_english = []
apple_english = []

for app in android_clean : 
    name = app[0]
    
    if english_detector(name) : 
        android_english.append(app)
    
for app in apple_apps : 
    name = app[1]
    
    if english_detector(name) : 
        apple_english.append(app)
        
print(f'After length : {len(android_english), len(apple_english)}')

Before length : (9659, 7197)
After length : (9614, 6183)


In [16]:
print(explore_data(android_english, 0, 3, rows_and_columns = True))
print('\n')
print(explore_data(apple_english, 0, 3, rows_and_columns = True))

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9614
Number of columns: 13
None


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '

Now we get datasets without any wrong data, duplicate data, non-english data. Android dataset have 9615 apps, Apple datasets have 6183 apps.

## Isolating the free apps

In android_english, index number of 'pricing' is 7, while apple_english is 4.

In [17]:
android_free = []
apple_free = []

for app in android_english :    
    price = app[7]      # We need to use '0' because character '$' is in front of price. 
    
    if price == '0' : 
        android_free.append(app)
        
for app in apple_english : 
    price = app[4]
    
    if price == '0.0' : 
        apple_free.append(app)
        
print(f'length of android_free : {len(android_free)}\nlength of apple_free : {len(apple_free)}')

length of android_free : 8864
length of apple_free : 3222


The final length of datasets for analysis are 8864 of android_free and 3222 of apple_free. 

## Most common genres 

So far, we spen a good amount of time on cleaning data, and : 

- Removed inaccurate data
- Removed duplicate app entries
- Removed non-English apps
- Isolated the free apps 

As we mentioned in the introduction, our aim is to determine the kinds of apps that are likely to attract more users because our revenue is highly influenced by the number of people using our apps.

To minimize risks and overhead, our validation strategy for an app idea is comprised of three steps : 

1. Build a minimal Android version of the app, and add it to Google play.
2. If hte app has a good respones from users, we develop it further.
3. If the app is profitable after six months, we build an iOS version of the app and add it to the App Store. 

Our conclusion is that we'll need to build a frequency table for the 'prime_genre' column of the apple_free dataset, and for the 'Genre' and 'Category' columns of the android_free dataset. 

We'll build two functions we can use to analyze the frequency tables : 

- One function to generate frequency tables that show percentages
- Another function we can use to display the percentages in a descending order 

In [18]:
# Create frequency table

def freq_table(dataset, index) : 
    
    freq_dict = {}        # Make dict storing counts grouped by index 
    
    for app in dataset : 
        want_col = app[index] 
        
        if want_col in freq_dict : 
            freq_dict[want_col] += 1
        else : 
            freq_dict[want_col] = 1
            
    freq_percentage = {}        # Store percentage of freq_dict
    
    for key in freq_dict : 
        percentage = (freq_dict[key] / len(dataset)) * 100 
        freq_percentage[key] = percentage
    
    return freq_percentage

def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

### Apple Store

In [19]:
display_table(apple_free, 11)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


We can see that about 58% of the apps in App Store are Games. Entertainment apps have about 8%, while Photo $ Video have only 5%. 

The general impression is that engertainment(games, photh and video, social networking, sports, music) apps have large cluster in App Store, while practical purposes(education, shopping, utilities, productivility, life style) apps have smaller cluster. But this result doesn't imply that apps of that genre generally have a large number of users. It's just count of offering. 

### Google Play

In [20]:
# Analysis from column name'Category'

display_table(android_free, 1)

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

In [21]:
# Analysis from column name 'genre'

display_table(android_free, 9)

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

In the result of display_table(android_free, 9), the most common genres is 'Tools' which have about 8.5%. The second most common genres in Google Play is 'Entertainmen'(about 6%), followed by 'Education'(about 5%)
We can see that apps in Google Play have same proportion of entertainment and practical purpose, while apps in Apple Store have more proportion of entertaiment. 

It is incorrect thinking that the genre with more frequency will attract more users. So we need to get an idea about the kinds of apps that have most users for our goal. 

## Most popular genres

One way to find out what genres are the most popular is to calculate the average number of installs for each genre. For the Google Play dataset, we can find this information in the 'Installs' column, but this information is missig for App Store dataset. As a workaround, we'll take the total number of user ratings as a proxy, which we can find in the 'rating_count_tot' app. 

### Apple Store

In [24]:
genres_apple = freq_table(apple_free, 11)

for genre in genres_apple : 
    total = 0    # store the sum of user ratings
    len_genre = 0    # store the number of apps specific to each genre 
    
    for app in apple_free :
        genre_app = app[11] 
            
        if genre_app == genre : 
            n_ratings = float(app[5])
            total += n_ratings 
            len_genre += 1
            
    avg_n_ratings = total / len_genre
    print(f'{genre} : {avg_n_ratings}')

Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Games : 22788.6696905016
Music : 57326.530303030304
Reference : 74942.11111111111
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Utilities : 18684.456790123455
Travel : 28243.8
Shopping : 26919.690476190477
News : 21248.023255813954
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 39758.5
Finance : 31467.944444444445
Education : 7003.983050847458
Productivity : 21028.410714285714
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


| Rank | Most common | Most popular | 
|:---:|:---:|:---:|
|1|Games|Navigation|
|2|Entertainment|Reference|
|3|Photo & Video|Social Networking|
|4|Education|Music|
|5|Social Networking|Weather| 
|6|Shopping|Book|
|7|Utilities|Food & Drink|
|8|Sports|Finance|
|9|Music|Photo $ Video|
|10|Health & Fitness|Travel|
|11|Productivity|Shopping|
|12|Lifestyle|Health & Fitness|
|13|News|Sports|
|14|Travel|Gamess|
|15|Finance|News|
|16|Weather|Productivity|
|17|Food & Drink|Utilities|
|18|Reference|Lifestyle|
|19|Business|Entertainment|
|20|Book|Business|
|21|Navigation|Education|
|22|Medical|Catalogs|
|23|Catalogs|Medical|


In the previous analysis, the most common genres in Apple Store, entertainment genre are the most common genre. However when we saw above chart, practical purpose genre are more pupular than entertainment genre. 


To accomplish our goal, we need to choos genre which is not common but popular. It means genre doesn't have many competitor, but have many users. But there is some missing point calculating rank by mean. mean is very sensitive statistics affected by outliers. So, we need to have deeper investigation.

Anyway, the most attractive genre will have huge difference gap between common and popluarity.(lower common genre rank and higher poular genre rank will be good!). They will be Navigation and Reference. 

### Google Play

We have data about the number of installs for the Google Play market, so we should be able to get a clearer picture about genre popularity. However, the isntall numbers don't seem precise enough - we can see that most values are open-ended.

In [26]:
display_table(android_free, 5)

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


For instance, we don' know whether an app with 100,000 + installs has 100,000 installs, 200,000, or 350,000. However, we don't need very precise data for our purposes - we only want to find out which app genres attract the most users. 

We're going to leave the numbers as they are, which means that we'll consider that an app with 100,000+ installs has 100,000 installs, and an app with 1,000,000_ installs has 1,000,000 installs. 

In [30]:
categories_android = freq_table(android_free, 1)

for category in categories_android : 
    
    total = 0
    len_category = 0
    
    for app in android_free : 
        category_app = app[1]
        
        if category_app == category : 
            n_installs = app[5]
            n_installs = n_installs.replace(',', '')
            n_installs = n_installs.replace('+', '')
            total += float(n_installs)
            len_category += 1
    avg_n_installs = total / len_category
    print(f'{category} : {avg_n_installs}')

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

|RANK| Most common | Most popular | 
|:--:|:---:|:---:|
|1|FAMILY | COMMUNICATION | 
|2|GAME | VIDEO_PLAYERS | 
|3|TOOLS | SOCIAL | 
|4|BUSINESS | PHTOGRAPH | 
|5|LIFESTYLE | PRODUCTIVITY | 
|6|PRODUCTIVITY | GAME |
|7|FINANCE | TRAVEL_AND_LOCAL | 
|8|MEDICAL | ENTERTAINMENT | 
|9|SPORTS | TOOLS | 
|10|PERSONALIZATION | NEWS_AND_MAGAZINE | 
|11|COMMUNICATION | BOOKS_AND_REFRENCE | 
|12|HEALTH_AND_FITNESS | SHOPPING | 
|13|PHOTOGRAPHY | PERSONALIZATION |
|14|NEWS_AND_MAGAZINES | WEATHER |
|15|SOCIAL |HEALTH_AND_FITNESS|
|16|TRAVEL_AND_LOCAL | MAPS_AND_NAVIGATION |
|17|SHOPPING | FAMILY|
|18|BOOKS_AND_REFERENCE | SPORTS|
|29|DATING | ART_AND_DESIGN|
|20|VIDEO_PLAYERS | FOOD_AND_DRINK|
|21|MAPS_AND_NAVIGATION | EDUCATION |
|22|FOOD_AND_DRINK | BUSINESS |
|23|EDUCATION | LIFESTYLE | 
|24|ENTERTAINMENT | FINANCE |
|25|LIBRARIES_AND_DEMO | HOUSE AND HOME |
|26|AUTO_AND_VEHICLES| DATING |
|27|HOUSE_AND_HOME | COMICS |
|28|WEATHER | AUTO_AND_DEMO|
|29|EVENTS | LIBRARIES_AND_DEMO |
|30|PARENTING | PARENTING |
|31|ART_AND_DESIGN | PARENTING |
|32|COMICS | EVENT |
|33|BEAUTY | MEDICAL |



The most popular genres in Google Play are 'COMMUNICATION', 'VIDEO_PLAYERS', 'SOCIAL'. But these genres have strong apps such as Facebook, Youtube, Twitter. So we need to find other genres which is potential.(lower common and higher popular). 

The 'BOOKS_AND_REFRENCE' will be potential for our goal. 

## Conclusions

In this project, we analyzed data about the App Store and Google Play mobile apps with the goal of recommending an app profile that can be profitable for both markets.

We analyze by checking most common genres and most popular genres to find those apps. The lower common genre and the higher popular genre will be attractive genres for our goal, books and reference genre will be result. 